### PR TITLE
Path root

### PR DESCRIPTION
**Pull request by tonglil**
Add ability to add a root to all paths.

Use case: we want to allow all users to cache in a shared `/drone-cache/*` bucket in S3 and not key off of `/<owner>/*`.

### CODE REVIEW

**Comment by tonglil** [comment]
PTAL @tboerger @appleboy @jmccann

**Comment by donny-dont** [comment]
@tboerger can you kick appveyor or just go ahead and merge this. Seems fine to me.

**Comment by donny-dont** [comment]
Grr looks like we need the other PR to merge first.

**Comment by tboerger** [comment]
Looks like wrong packages have been imported?

File: main.go [hunk 1]
```diff
@@ -285,3 +300,7 @@ func isMultipleModes(bools ...bool) bool {
 
 	return false
 }
+
+func prefixRoot(root, path string) string {
+	return pathutil.Clean(fmt.Sprintf("/%s/%s", root, path))
```
Comment (donny-dont) [inline main.go]: This should be path. I think that's why its not building.

Comment (tonglil) [reply main.go]: Whoops, I forgot to add the import alias.

Can't use path, it's already used.

Comment (donny-dont) [reply main.go]: Tried a couple commits to get it going. Feel free to squash or whatever @tonglil

Comment (donny-dont) [reply main.go]: Actually the build is working. Looks like the merge commit might have been stale.

**Comment by donny-dont** [comment]
🎊 

Not sure why Drone glitched on the PR but things seem good to go.

**Comment by tonglil** [comment]
Thanks!

**Pull request merged**